python/tests/illarg_tests/arr_map_type_fn.py

```diff
@@ -0,0 +1,674 @@
+from tests.aggregate_tests.aggtst_base import TstView
+
+
+# INDEX function
+class illarg_index_legal(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"arr": "14", "arr2": "See you!"}]
+        self.sql = """CREATE MATERIALIZED VIEW index_legal AS SELECT
+                      arr[2] AS arr,
+                      arr[SAFE_OFFSET(2)] AS arr2
+                      FROM illegal_tbl
+                      WHERE id = 0"""
+
+
+# Negative Test
+class illarg_index_illegal(TstView):
+    def __init__(self):
+        # checked manually
+        self.sql = """CREATE MATERIALIZED VIEW index_illegal AS SELECT
+                      bin[2] AS bin
+                      FROM illegal_tbl"""
+        self.expected_error = "Cannot apply 'ITEM' to arguments of type"
```

Comment:
frankly this is not a great error message, but that's how it comes from Calcite... who knows what ITEM is?
Can you please file an issue so we can improve it?